php_phongo.c

```diff
@@ -1934,9 +1945,10 @@ static inline char* php_phongo_fetch_ssl_opt_string(zval* zoptions, const char*
 	return value;
 }
 
-static mongoc_ssl_opt_t* php_phongo_make_ssl_opt(zval* zoptions TSRMLS_DC)
+static mongoc_ssl_opt_t* php_phongo_make_ssl_opt(mongoc_uri_t* uri, zval* zoptions TSRMLS_DC)
 {
 	mongoc_ssl_opt_t* ssl_opt;
+	bool              any_ssl_option_set = false;
```

Comment:
We still need any_ssl_option_set here, to see whether any changes were actually made. And if not, we need to deallocate the new ssl_opt and return NULL, so that we don't inadvertently end up setting an empty SSL opt struct on the client (that would turn SSL on).

Technically, that possibly means we don't have to do the ``mongoc_client_get_ssl`` check at https://github.com/mongodb/mongo-php-driver/pull/893/files#diff-c06c6e1c9374aecabcf544157f9d0c26R2479 any more, but I left that for now.

Reply:
Another alternative would be declaring `mongoc_ssl_opt_t empty_ssl_opt = {0}` up top and later checking if `memcmp(ssl_opt, &empty_ssl_ssl_opt, sizeof(mongoc_ssl_opt_t))` returns zero later to check if any modifications have been made.

Reply:
Yeah, that's an alternative, but not one I like. :-þ